chat-server/src/main/java/server/SQLiteService.java

```diff
@@ -0,0 +1,134 @@
+                "(SELECT id FROM Users WHERE nickname = ?), " +
+                "(SELECT id FROM Users WHERE nickname = ?), " +
+                " ?)");
+    }
+
+    public static void disconnect() {
+        try {
+            psChangeNickname.close();
+            psRegistration.close();
+            psGetNickname.close();
+//            psGetIdNickname.close();
+            psAddMessage.close();
+            psGetMessageForNick.close();
+        } catch (SQLException throwables) {
+            throwables.printStackTrace();
+        }
+        try {
+            connection.close();
+        } catch (SQLException throwables) {
+            throwables.printStackTrace();
+        }
+    }
+
+    public static String getNicknameByLoginAndPassword(String login, String password) {
+        try {
+            psGetNickname.setString(1, login);
+            psGetNickname.setString(2, password);
+            ResultSet resultSet = psGetNickname.executeQuery();
+            if (resultSet.next()) return resultSet.getString("nickname");
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    public static boolean registration(String login, String password, String nickname) {
+        try {
+            psRegistration.setString(1, login);
+            psRegistration.setString(2, password);
+            psRegistration.setString(3, nickname);
+            if (psRegistration.executeUpdate() > 0) return true;
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return false;
+    }
+
+    public static boolean changeNick(String oldNick, String newNick) {
+        try {
+            psChangeNickname.setString(1, newNick);
+            psChangeNickname.setString(2, oldNick);
+            if (psChangeNickname.executeUpdate() > 0) return true;
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return false;
+    }
+
+    public static boolean writeMessage(String senderNick, String receiverNick, String message) {
+        try {
+            psAddMessage.setString(1, senderNick);
+            psAddMessage.setString(2, receiverNick);
+            psAddMessage.setString(3, message);
+            if (psAddMessage.executeUpdate() > 0) return true;
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return false;
+    }
+
+    public static String getMessagesFromDB(String nickname) {
+        StringBuilder sb = new StringBuilder();
+        try {
+            psGetMessageForNick.setString(1, nickname);
+            ResultSet rs = psGetMessageForNick.executeQuery();
+            while (rs.next()) {
+                String senderNick = rs.getString(1);
+                String receiverNick = rs.getString(2);
+                String message = rs.getString("message");
+                if (receiverNick == null) {
+                    sb.append(String.format("[ %s ]: %s\n", senderNick, message));
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.